common/lib/capa/capa/capa_problem.py

```diff
@@ -855,17 +870,67 @@ def _preprocess_problem(self, tree):  # private
             for entry in inputfields:
                 entry.attrib['response_id'] = str(response_id)
                 entry.attrib['answer_id'] = str(answer_id)
                 entry.attrib['id'] = "%s_%i_%i" % (self.problem_id, response_id, answer_id)
                 answer_id = answer_id + 1
 
-            # Find the label and save it for html transformation step
-            responsetype_label = response.find('label')
-            problem_data[self.problem_id + '_' + str(response_id)] = {
-                'label': responsetype_label.text if responsetype_label is not None else ''
+            question_id = self.problem_id + '_' + str(response_id)
+            label = ''
+            element_to_be_deleted = None
+
+            # Extract label value from <label> tag or label attribute from inside the responsetype
+            # Extract label value from <label> tag
+            responsetype_label_tag = response.find('label')
+            if responsetype_label_tag is not None:
+                label = responsetype_label_tag.text
+                # Delete <label> tag containing question text otherwise question will be rendered twice
+                element_to_be_deleted = responsetype_label_tag
+            elif 'label' in inputfields[0].attrib:
+                # Extract label value from label attribute
+                # This is the case when we have a problem with multiple questions having OLX only
+                label = inputfields[0].attrib['label']
+                # Delete <p> tag containing question text otherwise question will be rendered twice
+                p_tag = response.xpath("preceding-sibling::p[text()='{}']".format(label))
+                if p_tag:
+                    element_to_be_deleted = p_tag[0]
+            else:
+                # neither <label> tag nor label attribute is present inside responsetype
+                # existing problem with multi-questions without --- having markdown
+                # find the immediate <label> tag before the responsetype. also delete it avoid rendering twice
+                label_tag = response.xpath("preceding-sibling::label[1]")
+                if label_tag:
+                    label = label_tag[0].text
+                    element_to_be_deleted = label_tag[0]
+
+            label = label.strip() or DEFAULT_QUESTION_TEXT
+
+            # delete label or p element only if responsetype is fully accessible
+            if response.tag in ACCESSIBLE_CAPA_RESPONSE_TYPES and element_to_be_deleted is not None:
+                element_to_be_deleted.getparent().remove(element_to_be_deleted)
+
+            # for non-accessible responsetypes it may be possible that label attribute is not present
+            # in this case pass an empty label. remember label attribute is only used as value for aria-label
+            if response.tag not in ACCESSIBLE_CAPA_RESPONSE_TYPES and label == DEFAULT_QUESTION_TEXT:
+                label = ''
```

Comment:
@cahrens Question label will never be empty for fully accessible problems. We will set it to default value if its empty. The above check is only used for problems which are not fully-accessible.